sbws/util/config.py

```diff
@@ -21,57 +25,87 @@
     conf = ConfigParser(interpolation=ExtendedInterpolation())
-    fname = os.path.join(PKG_DIR, 'config.default.ini')
-    assert os.path.isfile(fname)
-    conf = _read_config_file(conf, fname)
-    return conf
+    return _extend_config(conf, DEFAULT_CONFIG_PATH)
 
 
 def _get_user_config(args, conf=None):
+    """Get user configuration.
+    If a custom path is passed, search for user configuration there.
+    In any case, create a minimal user configuration in user configuration path
+    , in case it needs to be overwritten.
+    """
     if not conf:
         conf = ConfigParser(interpolation=ExtendedInterpolation())
     else:
         assert isinstance(conf, ConfigParser)
-    fname = os.path.join(args.directory, 'config.ini')
-    if not os.path.isfile(fname):
+    if args.config:
+        # if config is specified
+        # if it exits: use it
+        # if it does not exits: create it in the path provided
+        # normally, the config would not be created if it doesn't exits, but
+        # it's being created here so that it's easier to the user to know
+        # which options should be customized, since there're a lot
+        if os.path.isfile(args.config):
+            log.info("Using configuration file %s", args.config)
+            return _extend_config(conf, args.config)
+        _create_user_config_file(args.config)
         return conf
-    conf = _read_config_file(conf, fname)
-    return conf
+    # if config file is not specified: create user config in default path
+    # normally, the config would not be created if it doesn't exits, but
+    # it's being created here so that it's easier to the user to know
+    # which options should be customized, since there're a lot
+    _create_user_config_file(USER_CONFIG_PATH)
+    return _extend_config(conf, USER_CONFIG_PATH)
 
 
-def _get_default_logging_config(args, conf=None):
+def _get_default_logging_config(conf=None):
+    """Get default logging configuration."""
     if not conf:
         conf = ConfigParser(interpolation=ExtendedInterpolation())
     else:
         assert isinstance(conf, ConfigParser)
-    fname = os.path.join(PKG_DIR, 'config.log.default.ini')
-    assert os.path.isfile(fname)
-    conf = _read_config_file(conf, fname)
-    return conf
+    return _extend_config(conf, DEFAULT_LOG_CONFIG_PATH)
 
 
 def get_config(args):
+    """Get ConfigParser interpolating all configuration files."""
     conf = _get_default_config()
-    conf = _get_default_logging_config(args, conf=conf)
+    conf = _get_default_logging_config(conf=conf)
     conf = _get_user_config(args, conf=conf)
-    return conf
-
-
-def get_user_example_config():
-    conf = ConfigParser(interpolation=ExtendedInterpolation())
-    fname = os.path.join(PKG_DIR, 'config.example.ini')
-    assert os.path.isfile(fname)
-    conf = _read_config_file(conf, fname)
+    # it is only needed to expand user and vars to have the rest of the
+    # paths correctly expanded too
```

Comment:
To elaborate a little more on why I like the "beautiful solution" the most: this comment is true for the default configuration options as they exist today and is true for how I expect users to use sbws. But other than that, it isn't necessarily true. I don't think there's anything stopping users from configuring paths in such a way that this comment becomes false.